iceoryx_posh/test/mocks/roudi_memory_provider_mock.hpp

```diff
@@ -56,14 +62,19 @@ class MemoryProviderTestImpl : public iox::roudi::MemoryProvider
             destroyMemoryMock();
         }
 
-        free(dummyMemory);
-        dummyMemory = nullptr;
+        if (dummyMemory != nullptr)
```

Comment:
this check is superfluous and just adds visual noise

Reply:
fixed